integration-cli/docker_cli_build_test.go

```diff
@@ -7292,3 +7297,25 @@ func (s *DockerSuite) TestBuildWindowsEnvCaseInsensitive(c *check.C) {
 		c.Fatalf("Case insensitive environment variables on Windows failed. Got %s", res)
 	}
 }
+
+// GETENV on Windows #29048
+func (s *DockerSuite) TestBuildWindowsGetenv(c *check.C) {
+	testRequires(c, DaemonIsWindows)
+	name := "testbuildwindowsgetenv"
+	_, out, err := buildImageWithOut(name, `
+		FROM `+WindowsBaseImage+`
+		GETENV PATH
+		ENV PATH c:\\something;$PATH
+		RUN echo Result:%PATH%
+  `, true)
+	if err != nil {
+		c.Fatal(err)
+	}
+	c.Assert(strings.ToLower(out), checker.Contains, `result:c:\something;`)
+	c.Assert(strings.ToLower(out), checker.Contains, `c:\windows\system32`)
```

Comment:
I did it explicitly as two checks as I don't want to take a dependency on the original value of the path which may a) change in the future and b) be different in different base images (currently tests run on windowsservercore, and although nanoservers default path is identical today, that may change). So hence, I'm looking expressly for c:\windows\system32 which is a common place and unlikely to change; and for the inclusion of the new path. Coding something like that flexibly and simply inside the dockerfile isn't as easy to do without a mini-powershell script.